test/database_service/test_db_experiment_data.py

```diff
@@ -953,7 +954,7 @@ def callback3(exp_data):
 
         self.assertEqual(exp_data.analysis_status(), AnalysisStatus.ERROR)
         self.assertTrue("RuntimeError: YOU FAIL" in exp_data.analysis_errors())
-        self.assertEqual(len(results), 2)
+        self.assertEqual(len(results), 1)
```

Comment:
Why this is changed?

Reply:
The test fails before this change. It cancels the second out of three analyses, and checks how many analyses have run.

Previously, the correct answer was 2: Analysis 1 and 3 were running, and only Analysis 2 (the cancelled analysis) did not run.

But now, if an analysis is cancelled, then all the analysis that come after it are also cancelled. We are left with only Analysis 1 running, and this is why the correct answer is updated to 1.

As discussed in #870, I'd actually prefer not to cancel analysis that come after a failed analysis, but this is a change of behavior that looks intentional and therefore requires discussion.

Reply:
Understand. Likely this is intentional behavior but test was wrong?

Reply:
This is what I think, it will be reverted when #870 is done.